TesSense.py

```diff
@@ -33,142 +37,208 @@ def printerror(error, err) :                          # Error message with trunc
 sense = Senseable(wss_timeout=30,api_timeout=30)
 sense.authenticate(username, password)
 
+# Python3 -m pip install senselink
+from senselink import SenseLink
+
 # Python3 -m pip install teslapy
 print ("Starting connection to Tesla...")
 import teslapy
-with teslapy.Tesla( username ) as tesla:
-    vehicles = tesla.vehicle_list()
-    car = vehicles[0].get_vehicle_summary()
-    print(car['display_name'], "is", car['state'], "\n")
-    
-    #print( "Tesla Info:\n", vehicles[0].get_vehicle_data(), "\n\n" ) # shows all car data available
-    #print( "TeslaPy:\n", dir( teslapy ), "\n\n" )         # shows all the TeslaPy API functions
-    #print( "Senseable:\n", dir( Senseable ), "\n\n" )     # shows all the Sense API functions
-
-    while (True):
-        try :
-            sense.update_realtime()                   # Update Sense info
-            asp = int(sense.active_solar_power)
-            ap = int(sense.active_power)
-            power_diff = asp-ap                       # Watts being set back to the grid
-        except :
-            printmsg("Sense Timeout")
-            sleep(10)
-            continue # back to the top of the order
-
-        if vehicles[0].available() :
+
+
+async def run_tes_sense(mutable_plug):
+    rate = minrate = 0  # Minimum rate you want to set the charger to
+    volts = 120  # Minimum volts, until detected by the charger
+    pshown = loop = charge = charging = False  # init more variables
+
+    with teslapy.Tesla( username ) as tesla:
+        if not tesla.authorized:
+            tesla.refresh_token(refresh_token=input('Enter SSO refresh token: '))
```

Comment:
I use a refresh token to login to Tesla, remove these lines to go back to the original auth method